src/trade-params/TradeParams.js

```diff
@@ -85,6 +85,10 @@ export default class TradeParams extends Component {
         this.onStopTypeChange = ::this.onStopTypeChange;
         this.onStopProfitChange = ::this.onStopProfitChange;
         this.onPurchase = ::this.onPurchase;
+
+        this.state = {
+            dynamicKey: 0,
+        };
```

Comment:
you mean another middle state? or you dont think controlled component will disallow user input

I'll reply both

### Why controlled Component does not work?
I am not 100% sure of the reason, I think it happens like this

1. User key in `1`
2. Component fire action
3. Component updated by receiving updated props
4. User key in `2`
5. go back to 2.

I think it has to do with chrome treating field in datetime as a whole, the whole field is always selected, as opposed to text field where you have a moving cursor.

### Why need another middle state ?
to handle the problem, we need to have a state which does not update store whenever user key in, but update the controlled component state, and then we might need to set some timer so that the user input finally propagate to store.

This sounds complicated enough in words.

Note: I just tried in firefox and the start later is not working at all, now I start to appreciate the dropdown in binary.com XD, we might need to use text input or some ui library >.<